app/registries/foreman/plugin/importer_registry.rb

```diff
@@ -0,0 +1,23 @@
+module Foreman
+  class Plugin
+    class ImporterRegistry
+      attr_accessor :importers 
+
+      def initialize
+        @importers = {
+          :puppet => 'PuppetFactImporter'
+        }.with_indifferent_access
+      end
+
+      def register(key, klass, default = false)
+        @importers.default = klass if default
```

Comment:
Changing to `@importers['default']` here would make indifferent_access not needed.

Reply:
@bastilian, it's a real [property](https://docs.ruby-lang.org/en/2.0.0/Hash.html#method-i-default) in `Hash`. It sets the value that will be returned if a key is not present in hash:

``` ruby
@importers.default = 'PuppetFactImporter'

@importers['zzz']
# => 'PuppetFactsImporter'
```

Reply:
@ShimShtein I had no idea about `default`. Shouldn't we in that case avoid it even more?

Reply:
@bastilian  agreed, I think we should avoid the `default`. Will still wait for @ShimShtein on this.